src/cpp/fastdds/builtin/typelookup/TypeLookupManager.cpp

```diff
@@ -348,14 +348,15 @@ bool TypeLookupManager::create_endpoints()
     // Built-in request writer
     if (builtin_protocols_->m_att.typelookup_config.use_client)
     {
+        request_listener_ = new TypeLookupRequestListener(this);
```

Comment:
On the creation of the request reader there is another statement just like this one. Since this is the first time, the `new` should be done here, but on the reader creation it should be changed to:
1. Checking if it's `nullptr` before calling `new` (else we'll have leaks)
2. Only call `delete` if the listener was created for the reader (meaning if it had been created for the writter then it should not be deleted there)

I think I'd refactor this method to:
1. Be single return
2. Have a `ret` boolean initialized to `true` and used as entry condition to each step, if something fails it should be changed to `false`.
3. At the very end check `ret` and only call delete if `false`